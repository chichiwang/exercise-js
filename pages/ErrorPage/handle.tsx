import type { PageHandle } from 'app/types';

const handle: PageHandle = {
  head: {
    title: 'An Error Occurred',
  },
};

export default handle;
