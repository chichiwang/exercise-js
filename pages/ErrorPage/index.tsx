import React from 'react';
import { Link } from 'react-router-dom';

export default function ErrorPage() {
  return (
    <>
      <h1>An Error Occurred!</h1>
      <Link to="/">Home</Link>
    </>
  );
}
