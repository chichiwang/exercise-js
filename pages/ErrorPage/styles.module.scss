@use 'sass:map';
@use 'app/var.module';

.logoBlock {
  display: flex;
  justify-content: center;
  margin: calc(map.get(var.$content, 'padding') * 4) auto;
}

.errorHeading,
.errorMessage {
  text-align: center;
}
