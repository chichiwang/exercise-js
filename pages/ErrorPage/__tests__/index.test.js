import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import ErrorPage from '../index';

describe('ErrorPage Component', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<ErrorPage />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('contains the error text', () => {
    render(<ErrorPage />);

    expect(screen.getByRole('heading')).toHaveTextContent('An Error Occurred!');
  });
});
