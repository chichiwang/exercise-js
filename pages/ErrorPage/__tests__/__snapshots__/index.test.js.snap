// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component matches snapshot 1`] = `
[
  <h1>
    An Error Occurred!
  </h1>,
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>,
]
`;
